sherpa/plot/__init__.py

```diff
@@ -341,12 +430,14 @@ def _merge_settings(self, kwargs):
         """Return the plot preferences merged with user settings."""
         return {**self.contour_prefs, **kwargs}
 
-    def contour(self, x0, x1, y, levels=None, title=None, xlabel=None,
+    def contour(self, x0, x1, y, title=None, xlabel=None,
```

Comment:
I don't understand why the `levels` argument has been removed since I'd think it is an essential part of what a  "contour plot" is? However, thinking about it, we don't let users control the levels used - e.g. (with CIAO 4.15) I can say (with the test data available)

```
sherpa-4.15.0> load_image("psf_0.0_00_bin1.img")
```

and then

```
sherpa-4.15.0> contour_data()

```

but we don't let the user change the levels - e.g. 

```
sherpa-4.15.0> contour_data(levels=[100, 1000, 10000, 100000])
TypeError: contour() got multiple values for argument 'levels'
```

So, this may be more of a potential RFE.

Reply:
Hmm, I see `DataContour` below, which subsets `Contour`, has a `levels` attribute and send it here, but the other classes don't seem to have it. My brain hurts.

Reply:
It's supposed to passed in as a keyword argument here, if it's used at all (not using it means default levels as determined by the plotting backend). The idea is (and that's certainly something we can discuss here, which I why I'm glad you are looking at this early even before all tests pass) that this functions (and similar ones like `Plot.plot` have the following, consistent signature: `func(self, data_in,  standard_set_of_axis_specifications, **kwargs)` where 
- data_in is e.g. x, y or x0, x1, y, ..., 
- standard_set_of_axis_specifications is xlabel, ylabel, overplot, clearwindow, and  
-  **kwargs catches everything else that's just passed through to the plotting backend.
 
You could argue that it's also a pretty integral part of `Plot.plot` to decide if you plot lines or points and yet `markertype` and `linestyle` are also relegated to `**kwargs`.

If we get to a more structure input, I hope we can summarize `data_in` in a single structure (maybe some dataclass, or maybe a DataSpace), then we have the standard set of how plots are done, and **kwargs for everything else. That means that the calling signature would be the same for e.g. `Plot.plot` and `Point.point` and `Histrogram.plot` etc. which means that the number of plot classes could be greatly reduced  and the decision on how it looks can be made mostly based on what he dataclass looks like (plot for X/Y data, point for X/Y data wiht just one entry, histgram of X_Lo/x_hi/y data, etc. 

It's not in the scope of this PT to make the changes (see #1375) but when I touch lines for some reason, then I have that in mind and reorder keyword argument to make that work (in this case, by removing "levels" which means it ends up in kwargs).

Reply:
Ta for the explainer.